pkg/doctype/doctype_test.go

```diff
@@ -0,0 +1,43 @@
+package doctype_test
+
+import (
+	"testing"
+
+	"github.com/alecthomas/assert"
+	"github.com/alecthomas/chroma/pkg/doctype"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDoctype_MatchString(t *testing.T) {
+	tests := map[string]struct {
+		Text     string
+		Pattern  string
+		Expected bool
+	}{
+		"simple html match": {
+			Text:     "<!DOCTYPE html> <html>",
+			Pattern:  `html.*`,
+			Expected: true,
+		},
+		"full html match": {
+			Text:     "<?xml ?><!DOCTYPE html PUBLIC  \"-//W3C//DTD XHTML 1.0 Strict//EN\">",
+			Pattern:  `html`,
+			Expected: true,
+		},
+		"xml not match": {
+			Text:     "<?xml ?> <DOCTYPE html PUBLIC \"a\"> <html>",
```

Comment:
This is not matching because of missing `!`. Not because it is an xml Docstring. Is this what you intended?

Reply:
Tests were copied from this piece of code https://github.com/pygments/pygments/blob/master/tests/test_util.py#L115

Reply:
Ok cool. Then let's rename test description from `xml not match` to `missing exclamation mark`